app/com/arpnetworking/kairos/client/KairosDbClientImpl.java

```diff
@@ -61,26 +64,25 @@
     @Override
     public CompletionStage<MetricsQueryResponse> queryMetrics(final MetricsQuery query) {
         final UUID queryUuid = UUID.randomUUID();
-        LOGGER.debug()
+        final JsonNode queryJson = _mapper.valueToTree(query);
+        LOGGER.trace()
                 .setMessage("starting queryMetrics")
                 .addData("queryUuid", queryUuid)
-                .addData("query", query)
+                .addData("query", queryJson)
                 .log();
-        try {
-            final HttpRequest request = HttpRequest.POST(createUri(METRICS_QUERY_PATH).toString())
-                    .withEntity(ContentTypes.APPLICATION_JSON, _mapper.writeValueAsString(query));
-            return fireRequest(request, MetricsQueryResponse.class)
-                    .whenComplete((response, error) -> {
-                        LOGGER.debug()
-                                .setMessage("finished queryMetrics")
-                                .addData("queryUuid", queryUuid)
-                                .addData("query", query)
-                                .addData("success", error == null)
-                                .log();
-                    });
-        } catch (final JsonProcessingException e) {
-            throw new RuntimeException(e);
-        }
+        final HttpRequest request = HttpRequest.POST(createUri(METRICS_QUERY_PATH).toString())
+                .withEntity(ContentTypes.APPLICATION_JSON, queryJson.toString());
+        final Instant startTime = Instant.now();
+        return fireRequest(request, MetricsQueryResponse.class)
+                .whenComplete((response, error) -> {
+                    LOGGER.trace()
+                            .setMessage("finished queryMetrics")
+                            .addData("queryUuid", queryUuid)
+                            .addData("query", queryJson)
+                            .addData("duration", Duration.between(startTime, Instant.now()))
```

Comment:
No problem logging it, but it seems like it should be emitted to metrics as well.

Reply:
Added!